newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js

```diff
@@ -82,15 +82,181 @@ export const declareBehaviorMetadata = (
  * events based object.
  */
 export const declareObjectMetadata = (
+  i18n: I18nType,
   extension: gdPlatformExtension,
   eventsBasedObject: gdEventsBasedObject
 ): gdObjectMetadata => {
-  return extension.addEventsBasedObject(
+  const objectMetadata = extension.addEventsBasedObject(
     eventsBasedObject.getName(),
     eventsBasedObject.getFullName() || eventsBasedObject.getName(),
     eventsBasedObject.getDescription(),
     getExtensionIconUrl(extension)
   );
+
+  // TODO EBO Use full type to identify object to avoid collision.
+  // Objects are identified by their name alone.
+  const objectType = eventsBasedObject.getName();
+
+  objectMetadata
```

Comment:
Nice! I wonder if this will be needed to be moved to a separate file at some point if this grow. For now it seems ok!